apps/utr-backend/src/endpoints/utils/AuthenticationUtils.ts

```diff
@@ -1,14 +1,30 @@
-import { DecodedIdToken, getAuth } from "firebase-admin/auth";
+import { getAuth } from "firebase-admin/auth";
 
-export async function isAuthenticated(req: any, res: any, next: any) {
+export async function isAuthenticatedAsAdmin(req: any, res: any, next: any) {
 	if (typeof req.headers.authorization !== "undefined") {
 		const token = req.headers.authorization.split(" ")[1];
 		try {
-			const decodeValue = await getAuth().verifyIdToken(token);
-			if (decodeValue) {
+			// GET ROLE -> save in req.role
+			await getAuth()
+				.verifyIdToken(token)
+				.then((claims: any) => {
+					if (
+						claims != undefined &&
+						claims.isAdmin != undefined &&
```

Comment:
Anstatt claims.isAdmin würde ich claims.role oder claims.userRole verwenden. Dadurch können wir die Anwendung in der Zukunft mit anderen Rollen erweitern.

Reply:
Damit könnte man auch die isAdmin middleware umschreiben, so dass diese auf req.user zugreift. Dadurch kann man sich das doppelte entschlüsseln des JWT sparen, wenn sowohl die getUser und die isAdmin middleware aufgerufen werden.

Reply:
Claim Name können wir gerne ändern. Schreibe ich morgen dann um.

Momentan war es so gedacht, dass die isAdmin middleware dort eingesetzt wird, wo nur Admins zugriff drauf haben dürfen (z.B. delete) und die getRole middleware dort wo Admins und User drauf zugreifen dürfen, aber je nach Rolle etwas anderes geschieht (z.B. create, beim Admin ist die Company direkt verifiziert).
Es würden also nie beide middlewares aufgerufen werden, somit auch kein doppeltes entschlüsseln stattfinden.
Wenn du meinst es ist besser eine middleware wie z.B. getRole zu machen und dann in der Route das Accesscontrol zu machen, ob es ein Admin, User oder unangemeldeter User ist und er die nutzen darf oder nicht, kann ich das auch gerne so umschreiben.

Reply:
Der Claimname ist jetzt role. Die genutzten Rollen sind im Moment user und admin. Wenn ein User eingeloggt ist, aber der Custom Claim role noch nicht bei ihm gesetzt ist, dann wird dieser bei ihm gesetzt, wenn die middleware getUser aufgerufen wird.